src/odemis/gui/comp/overlay/world.py

```diff
@@ -832,6 +833,606 @@ def deactivate(self):
+
+            # capture the mouse
+            self.cnvs.SetFocus()
+
+        else:
+            # Nothing to do with rulers
+            evt.Skip()
+
+    def _get_ruler_below(self, vpos):
+        """
+        Find a ruler corresponding to the given mouse position.
+        Args:
+            vpos (int, int): position of the mouse in view coordinate
+        Returns: (Ruler or None, HOVER_*): the most appropriate ruler and the hover mode.
+            If no ruler is found, it returns None.
+        """
+
+        if self._rulers:
+            for ruler in self._rulers:
+                hover_mode = ruler.get_hover(vpos)
+                if hover_mode != gui.HOVER_NONE:
+                    return ruler, hover_mode
+
+        return None, gui.HOVER_NONE
+
+    def on_motion(self, evt):
+        """ Process drag motion if enabled, otherwise call super method so event will propagate """
+        if not self.active:
+            return super(RulerOverlay, self).on_motion(evt)
+
+        vpos = evt.Position
+        if self._left_dragging:
+            if not self._selected_ruler:
+                logging.error("Dragging without selected ruler")
+                evt.Skip()
+                return
+
+            if evt.ControlDown():
+                ctrl_down = True
+            else:
+                ctrl_down = False
+            self._selected_ruler.on_motion(vpos, ctrl_down)
+            self.cnvs.request_drawing_update()
+        else:
+            # Hover-only => only update the cursor based on what could happen
+            _, drag = self._get_ruler_below(vpos)
+            if drag != gui.HOVER_NONE:
+                self.cnvs.set_dynamic_cursor(wx.CURSOR_PENCIL)
+            else:
+                self.cnvs.reset_dynamic_cursor()
+
+            evt.Skip()
+
+    def on_char(self, evt):
+        """ Delete the selected ruler"""
+        if not self.active:
+            return super(RulerOverlay, self).on_char(evt)
+
+        if evt.GetKeyCode() == wx.WXK_DELETE:
+            if not self._selected_ruler:
+                logging.debug("Deleted pressed but no selected ruler")
+                evt.Skip()
+                return
+
+            self._rulers.remove(self._selected_ruler)
+            self._selected_ruler = None
+            self.cnvs.request_drawing_update()
+        else:
+            evt.Skip()
+
+    def on_left_up(self, evt):
+        """ Stop drawing a selected ruler if the overlay is active """
+
+        if not self.active:
+            return super(RulerOverlay, self).on_left_up(evt)
+
+        if self._left_dragging:
+            if self._selected_ruler:
+                self._selected_ruler.stop_updating_ruler()
+                self.cnvs.update_drawing()
+
+            if self.ruler_mode == MODE_CREATE_RULER:
+                # Revert to the standard (NONE) tool
+                self.ruler_mode = MODE_SHOW_RULER
+                self._selected_tool_va.value = TOOL_NONE
+                self.active = True
```

Comment:
If self.active is False, the line will never be reached (due to the `if` at the beginning of this function). => drop this line.